### PR TITLE
Users can cancel running tasks

### PR DESCRIPTION
**Pull request by paulbalaji**
#### Description

- added a cancel button next to the info text for running tasks
- cancel button triggers editor dialog to cancel task or keep it running
- authentication tasks can't be cancelled by users (cancel button is hidden)

image

image

image

#### Tests

- tested cancelling the different possible tasks (upload assembly, launch dpl, stop dpl, refresh)

#### Documentation

tbd in final PR

#### Primary reviewers
If your change will take a long time to review, you can name at most two primary reviewers who are ultimately responsible for reviewing this request. @ mention them.

### CODE REVIEW

File: workers/unity/Packages/com.improbable.gdk.deploymentlauncher/Commands/WrappedTask.cs [hunk 1]
```diff
@@ -21,6 +21,11 @@ public void Dispose()
             CancelSource?.Dispose();
         }
 
+        public int GetId()
+        {
+            return Task?.Id ?? 0;
```
Comment (jessicafalk) [inline workers/unity/Packages/com.improbable.gdk.deploymentlauncher/Commands/WrappedTask.cs]: can 0 be a valid task id?

File: workers/unity/Packages/com.improbable.gdk.deploymentlauncher/DeploymentLauncherWindow.cs [hunk 2]
```diff
@@ -303,14 +303,68 @@ private void OnGUI()
...
+                    sb.Append("Are you sure you want to cancel launching your deployment?");
+                    break;
+                case ListTask task:
+                    sb.Append("Are you sure you want to cancel listing live deployments?");
+                    break;
+                case StopTask task:
+                    sb.Append("Are you sure you want to cancel stopping your deployment?");
+                    break;
+                case AuthTask task:
+                    Debug.Log("Cannot cancel an authenticated task.");
```
Comment (jessicafalk) [inline workers/unity/Packages/com.improbable.gdk.deploymentlauncher/DeploymentLauncherWindow.cs]:
```suggestion
                    Debug.Log("Cannot cancel authentication against SpatialOS.");
```

or is it possible to simply not show the cancel button for this?

Comment (paulbalaji) [reply workers/unity/Packages/com.improbable.gdk.deploymentlauncher/DeploymentLauncherWindow.cs]: yes that works better - I'll do that 👍

File: workers/unity/Packages/com.improbable.gdk.deploymentlauncher/Commands/WrappedTask.cs (hunk 1 above)
Comment (paulbalaji) [reply workers/unity/Packages/com.improbable.gdk.deploymentlauncher/Commands/WrappedTask.cs]: I can't find clarification here: https://docs.microsoft.com/en-us/dotnet/api/system.threading.tasks.task.id?view=netcore-2.2

But my understanding of this, 0 is not a valid task ID: https://blog.stephencleary.com/2013/03/a-few-words-on-taskid-and.html

Comment (jessicafalk) [reply workers/unity/Packages/com.improbable.gdk.deploymentlauncher/Commands/WrappedTask.cs]: that sounds good to me

**Pull request merged**